core/task.go

```diff
@@ -91,6 +91,15 @@ type Task interface {
 
 type TaskOption func(Task) TaskOption
 
+type A int
+
+type B struct {
+	foo int
+	bar int
+}
+
+type x func(int) x
+
```

Comment:
argh, I was trying to explain my coworker, how the TaskOption (undoing) works - the recursive func type - and forget to remove example - I'll delete it 
ps. is "undoing" feature used somewhere ? I couldn't find any usage - return value from calling is always ignored - or I'm missing something